src/storage/FileDescriptor.cpp

```diff
@@ -689,4 +688,26 @@ void FileDescriptor::setActualRights(uint64_t rights, uint64_t inheriting)
     rightsSet = true;
 }
 
+int FileDescriptor::duplicate(const FileDescriptor& other)
+{
+    // Duplicate the underlying fd
+    linuxFd = ::dup(other.linuxFd);
+
+    linuxMode = other.linuxMode;
+    linuxFlags = other.linuxFlags;
+    linuxErrno = other.linuxErrno;
+
+    // Set the internal variables
+    path = other.path;
+    rightsSet = other.rightsSet;
+    actualRightsBase = other.actualRightsBase;
+    actualRightsInheriting = other.actualRightsInheriting;
+
+    dirContentsLoaded = other.dirContentsLoaded;
+    dirContents = other.dirContents;
+    dirContentsIdx = other.dirContentsIdx;
+
+    return linuxFd;
```

Comment:
The implementation of `dup` was a bit of a hack before, and I _think_ it worked ok, but this is more explicit.